### DISCUSSION TITLE
suggestions in comments are often distracting

### DISCUSSION THREAD

**Discussion by verhovsky** [community/community#13007] [Copilot Conversations] [answered]
copilot loves trying to guess what I'm about to type in a comment, and usually I'm not sure what exactly I want to say in the comment or know exactly how I'm going to phrase it as I'm writing. So when it makes me read some AI-generated grammatically correct nonsense a few times as I'm gathering my thoughts it makes it harder, not easier.

It would nice if there could be a post processing step so that when a suggestion only contains commented out text and no source code, to not show the suggestion.

**Comment by transitive-bullshit** [comment, marked as answer]
Agreed. See also https://github.com/github/feedback/discussions/8062

**Comment by jamesonknutson** [comment]
Hello, I got tired of this as well, so I wrote a simple VSCode extension that turns off Copilot's `inlineSuggest` setting if your selection contains a TextMate scope that has the string `comment` in it.

[I wrote about it on the other issue @transitive-bullshit mentioned](https://github.com/github/feedback/discussions/8062#discussioncomment-2399108), I hope that you find it useful as a stop-gap until GH implements this officially.

**Comment by Lexian-droid** [comment]
I notice this as well, an option you can do is press ESC and it gets rid of the suggestion, even then though the bot will just keep suggesting code constantly, so it's not really that great of a solution. What would be cool is if they added some sort of syntax to tell it to stop trying to suggest code, it's also disruptive because when the AI is trying to constantly guess what you're looking for, it also breaks VSCode's auto completion a lot of the times.
